CMakeLists.txt

```diff
@@ -87,7 +85,7 @@ message (LDFLAGS="${LDFLAGS}")
 #------------------------------------------------------------------------
 #-- COMPILER OPTIONS
 #------------------------------------------------------------------------
-set(CMAKE_CXX_FLAGS "-O2 -Wall ${CMAKE_CXX_FLAGS} -std=c++17 -I./ ${LDFLAGS} ${REVCPU_COMPILER_MACROS}")
+set(CMAKE_CXX_FLAGS "-O2 -Wall -Wextra -Werror -Wuninitialized -Wmaybe-uninitialized -Wformat=2 -Wformat-overflow=2 -Wformat-signedness -Wformat-truncation=2 -Wuseless-cast -Wdouble-promotion -Wno-error=double-promotion -Wno-variadic-macros -Wno-error=variadic-macros -Wno-deprecated-declarations -Wvolatile -Wno-unused-variable -Wno-unused-parameter -Wno-shift-count-overflow ${CMAKE_CXX_FLAGS} -std=c++17 -I./ ${LDFLAGS} ${REVCPU_COMPILER_MACROS}")
```

Comment:
clang on macOS does not like some of these options:
`clang: error: -Wl,-undefined: 'linker' input unused [-Werror,-Wunused-command-line-argument]
clang: error: -Wl,dynamic_lookup: 'linker' input unused [-Werror,-Wunused-command-line-argument]`

Reply:
@donofrio: it should be fixed now.  I changed the options to only include ones which are compatible with both GCC and Clang.

Reply:
Looks like clang still does not like `-Werror` I'm running:

Apple clang version 14.0.0 (clang-1400.0.29.202)
Target: x86_64-apple-darwin22.6.0
Thread model: posix

Reply:
`-Werror` has been supported by Clang for many years. [Here's where the 14.0.0 docs mention it](https://releases.llvm.org/14.0.0/tools/clang/docs/UsersManual.html#options-to-control-error-and-warning-messages).

The `-W` options I give in CMake are unrelated to `-Wl,`. Your errors seem to be implying that you're specifying `-Wl,` options which are not being used if there is no linking (only compiling) going on. I don't know why Clang is taking `-W` warning options and turning them into `-Wl,` options.

**I would like a `-v` option added to `CMAKE_CXX_FLAGS`, so we can see how the compiler and linker are actually being invoked when you get your error (it makes the driver echo its subcommands).**

I suspect CMake more than Clang is the source of the problem, which is also why setting `CMAKE_CXX_STANDARD` to `17` didn't work for you. 

There's no reason my changing a CMake variable to add more `-W` warnings should cause that error, which seems to be interpreting `-W...` options as `-Wl,...` options instead. I wonder if our CMake code, or another script, has a regexp searching for `-W` options, and can't tell `-Werror` from `-Wl,-s`.

Reply:
This definitely fails on MacOS with Clang 13.0.0
```
[  6%] Building CXX object src/CMakeFiles/revcpu.dir/PanExec.cc.o
clang: error: -Wl,-undefined: 'linker' input unused [-Werror,-Wunused-command-line-argument]
clang: error: -Wl,dynamic_lookup: 'linker' input unused [-Werror,-Wunused-command-line-argument]
make[2]: *** [src/CMakeFiles/revcpu.dir/PanExec.cc.o] Error 1
make[1]: *** [src/CMakeFiles/revcpu.dir/all] Error 2
make: *** [all] Error 2
```

Reply:
This patch fixes the issue w/ a CMake guard: 
```
#------------------------------------------------------------------------
 #-- COMPILER OPTIONS
 #------------------------------------------------------------------------
-set(CMAKE_CXX_FLAGS "-std=c++17 -O2 -Wall -Wextra -Werror -Wvla -Wuninitialized -Wfloat-conversion -Wdouble-promotion -Wno-unused-parameter -Wno-deprecated-declarations ${CMAKE_CXX_FLAGS} -I./ ${LDFLAGS} ${REVCPU_COMPILER_MACROS}")
+if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
+  set(CMAKE_CXX_FLAGS "-std=c++17 -O2 -Wall -Wextra -Wvla -Wuninitialized -Wfloat-conversion -Wdouble-promotion -Wno-unused-parameter -Wno-deprecated-declarations ${CMAKE_CXX_FLAGS} -I./ ${LDFLAGS} ${REVCPU_COMPILER_MACROS}")
+else()
+  set(CMAKE_CXX_FLAGS "-std=c++17 -O2 -Wall -Wextra -Werror -Wvla -Wuninitialized -Wfloat-conversion -Wdouble-promotion -Wno-unused-parameter -Wno-deprecated-declarations ${CMAKE_CXX_FLAGS} -I./ ${LDFLAGS} ${REVCPU_COMPILER_MACROS}")
+endif()
 set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -O0 -Wall ${REVCPU_COMPILER_MACROS}")
 set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3 -Wall ${REVCPU_COMPILER_MACROS}")
```

Reply:
No `-Werror` for Clang? It seems strange that that would resolve "linker errors".

I'll merge it and the rest of the `devel` changes into my PR.